src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -403,30 +543,86 @@ public SymbolicMatch FindMatch(RegexRunnerMode mode, ReadOnlySpan<char> input, i
             }
             else
             {
-                Registers endRegisters = _pattern._info.ContainsLineAnchor ?
+                Registers endRegisters = _containsAnyAnchor ?
                     FindSubcaptures<FullInputReader>(input, matchStart, matchEnd, perThreadData) :
                     FindSubcaptures<NoZAnchorInputReader>(input, matchStart, matchEnd, perThreadData);
                 return new SymbolicMatch(matchStart, matchEnd - matchStart, endRegisters.CaptureStarts, endRegisters.CaptureEnds);
             }
         }
 
+        private int FindEndPositionOptimized<TOptimizedInputReader, TAcceleratedStateHandler, TOptimizedNullabilityHandler>(ReadOnlySpan<char> input, int pos, long timeoutOccursAt, RegexRunnerMode mode, PerThreadData perThreadData)
+            where TOptimizedInputReader : struct, IOptimizedInputReader
+            where TAcceleratedStateHandler : struct, IAcceleratedStateHandler
+            where TOptimizedNullabilityHandler : struct, IOptimizedNullabilityHandler
+        {
+            int initialStatePosCandidate = pos;
+            var currentState = new CurrentState(_dotstarredInitialStates[GetCharKind<FullInputReader>(input, pos - 1)]);
+            int endPos = NoMatchExists;
+
+            while (true)
+            {
+                const int CharsPerTimeoutCheck = 1_000;
+                // TODO: maybe this should be for NFA mode only
```

Comment:
?

Reply:
Actually this fine, but i think only NFA mode can reach speeds low enough for this limit to not cause overhead.
currently this is hit every 0.002mb, where the absolute worst DFA speed in traversal with no matches occurring is around 1gb/s.

I haven't investigated the overhead itself so this is just an assumption that the limit is maybe too low

Reply:
Did some testing on this:

303 matches in 900kb of text with timeout, pattern: `\w+gi+w`

- 100: 18.259 ms
- 1_000: 13.724 ms (currently)
- 5_000: 4.984 ms
- 10_000: 2.569 ms
- 15_000: 1.995 ms
- 25_000: 1.925 ms
- 100_000: 1.970 ms

15_000 or 25_000 should be reasonable here. If the timeout doesn't have to be reported too frequently then this could be raised for string literal prefixes too